dali/operators/reader/file_reader_op.cc

```diff
@@ -100,4 +100,20 @@ This argument is mutually exclusive with ``file_list``.)", nullptr)
 If not used, sequential 0-based indices are used as labels)", nullptr)
   .AddParent("LoaderBase");
 
+
+// Deprecated alias
+DALI_REGISTER_OPERATOR(FileReader, FileReader, CPU);
+
+DALI_SCHEMA(FileReader)
+    .DocStr("Legacy alias for :meth:`readers.file`.")
+    .NumInput(0)
+    .NumOutput(2)  // (Images, Labels)
+    .AddParent("readers__File")
+    .MakeDocPartiallyHidden()
+    .Deprecate(
+        "readers__File",
+        R"code(In DALI 1.0 all readers were moved into a dedicated :mod:`~nvidia.dali.fn.readers`
+submodule and renamed to fit a common pattern. This is a placeholder operator with identical
+functionality to allow for backward compatibility.)code");  // Deprecated in 1.0;
```

Comment:
Jest and idea - put a macro that does this in some common reader header? Then we'd have:
```suggestion
DALI_READER_ALIAS(File, file).
    .NumOutput(2)  // (Images, Labels)
```

The macro would be sth like:
```C++
#define DALI_READER_ALIAS(OpName, fn_name) \
  DALI_SCHEMA(OpName##Reader)
    .DocStr("Legacy alias for :meth:`readers." #fn_name "`.")
    .NumInput(0)
    .AddParent("readers__" #OpName)
    .MakeDocPartiallyHidden()
    .Deprecate(
        "readers__" #OpName,
        R"code(In DALI 1.0 all readers were moved into a dedicated :mod:`~nvidia.dali.fn.readers`
submodule and renamed to fit a common pattern. This is a placeholder operator with identical
functionality to allow for backward compatibility.)code");  // Deprecated in 1.0;
```

Reply:
I think I can consider this after we do decoders as well, and then it will be too complex of a macro.